src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: 720px;
+  gap: $gap;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    @include proportions();
+    box-sizing: border-box;
+
+    border: 1px solid #000;
+    background-color: #eee;
+
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: #ffc0cb;
+      cursor: pointer;
+      transform: translate(0, -20px);
+    }
+  }
+}
+
+@each $day, $first-day-shift in $days-of-the-week {
+  .calendar--start-day--#{$day}
+  .calendar__day:first-of-type {
+    margin-left: $first-day-shift;
+  };
+}
```

Comment:
@each $day, $first-day-shift in $days-of-the-week {
  .&--start-day--#{$day} &__day:first-of-type {
    margin-left: $first-day-shift;
  };

i think it`s better and you can move it to .calendar block in scss